### ISSUE TITLE
can not reproduce the results

### ISSUE THREAD

**Issue by zhixiaochuan12** [vered1986/LexNET#11]
Hello, I downloaded your pre-processed [corpus](https://drive.google.com/file/d/0B0kBcFEBhcbha2N0Vm1FYW01Umc/view?usp=sharing) files, and trained `train_integrated.py` on ROOT09 and SHWARTZ datasets. But both of them have pairs withour paths, thus not able to reproduce the results.

The details of pre-processed corpus files are:
<img width="295" alt="屏幕快照 2019-04-03 14 14 46" src="https://user-images.githubusercontent.com/16333419/55456920-e73a3200-561a-11e9-81e6-57248a39b030.png">

The commonds for ROOT09 is `python train_integrated.py -m 8192 resource/wiki ../datasets/ROOT09/ model_ ../glove.6B.50d.txt`. 

The results or ROOT09 are:

> [dynet] random seed: 3016748844
> [dynet] allocating memory: 8192MB
> [dynet] memory allocation done.
> Loading the dataset...
> Done!
> Loading the corpus...
> Done!
> Initializing word embeddings...
> Known lemmas: 84939 / 101427
> Loading path files...
> Pairs without paths: 3342 , all dataset: 12711
> Getting word vectors for the terms...
> Done loading corpus data!
> Number of words 416488, number of pos tags: 18, number of dependency labels: 47,
>  number of directions: 6
> Creating the network...
> Done!
> Training with learning rate = 0.001000, dropout = 0.000000...
> Training the model...
> The dy.parameter(...) call is now DEPRECATED.
>         There is no longer need to explicitly add parameters to the computation
> graph.
>         Any used parameter will be added automatically.
> Epoch 1 / 5 Loss = 0.957969462203
> Epoch 2 / 5 Loss = 0.691151392791
> Epoch 3 / 5 Loss = 0.583572979653
> Epoch 4 / 5 Loss = 0.49588360921
> Epoch 5 / 5 Loss = 0.427793715355
> Done!
> === Performance ===
> Mean precision:  0.735%
> Mean recall:     0.719%
> Mean F1:         0.724%
> Learning rate = 0.001000, dropout = 0.000000, Precision: 0.735, Recall: 0.719, F
> 1: 0.724
> Best hyper-parameters: Learning rate = 0.001000, dropout = 0.000000
> Saving the model...
> Evaluation:
>               precision    recall  f1-score   support
> 
>       RANDOM      0.781     0.703     0.740      1561
>        COORD      0.929     0.951     0.940       816
>        HYPER      0.543     0.635     0.585       803
> 
>    micro avg      0.750     0.750     0.750      3180
>    macro avg      0.751     0.763     0.755      3180
> weighted avg      0.759     0.750     0.752      3180
> 
> === Performance ===
> Mean precision:  0.759%
> Mean recall:     0.750%
> Mean F1:         0.752%
> Precision: 0.759, Recall: 0.750, F1: 0.752

The results of SHWARTZ datasets in hypenet project are:

> [dynet] random seed: 3016748844
> [dynet] allocating memory: 81920MB
> [dynet] memory allocation done.
> Loading the dataset...
> Done!
> Loading the corpus...
> Done!
> Initializing word embeddings...
> Known lemmas: 54586 / 65323
> Loading path files...
> Pairs without paths: 15561 , all dataset: 20314
> Getting word vectors for the terms...
> Done loading corpus data!
> Number of words 410737, number of pos tags: 18, number of dependency labels: 47,
>  number of directions: 6
> Creating the network...
> Done!
> Training with learning rate = 0.001000, dropout = 0.000000...
> Training the model...
> The dy.parameter(...) call is now DEPRECATED.
>         There is no longer need to explicitly add parameters to the computation
> graph.
>         Any used parameter will be added automatically.
> Epoch 1 / 5 Loss = 0.462441328414
> Epoch 2 / 5 Loss = 0.151701926858
> Epoch 3 / 5 Loss = 0.105326720303
> Epoch 4 / 5 Loss = 0.0825403603433
> Epoch 5 / 5 Loss = 0.0644293959151
> Done!
> === Performance ===
> Mean precision:  0.754%
> Mean recall:     0.771%
> Mean F1:         0.699%
> Learning rate = 0.001000, dropout = 0.000000, Precision: 0.754, Recall: 0.771, F
> 1: 0.699
> Best hyper-parameters: Learning rate = 0.001000, dropout = 0.000000
> Saving the model...
> Evaluation:
>               precision    recall  f1-score   support
> 
>         True      0.698     0.184     0.291      1234
>        False      0.769     0.972     0.859      3451
> 
>    micro avg      0.764     0.764     0.764      4685
>    macro avg      0.734     0.578     0.575      4685
> weighted avg      0.750     0.764     0.709      4685
> 
> === Performance ===
> Mean precision:  0.750%
> Mean recall:     0.764%
> Mean F1:         0.709%
> Precision: 0.750, Recall: 0.764, F1: 0.709

Did I miss something?  Thanks in advance.

**Comment by vered1986** [comment]
Hi @yangyuji12,

As I also wrote in the email (just now) - the HypeNET dataset contains rare words and terms. The LexNET corpus is constructed with a vocabulary of the 1/2 million most frequent terms in Wikipedia. So it's expected that there will be pairs with no paths. 

As for ROOT09, the number of pairs without paths is OK - unlike in HypeNET in which we created a dataset containing only pairs of words that occur together in the corpus, in LexNET we used existing datasets, so it happens that some words never occur together. 

Since the results are not that off, I suppose it's a matter of tuning (random seed, epochs, dropout rate, learning rate). 

I got the following performance: 

Validation: P = 0.817, R = 0.823, F1: 0.818
Test: P = 0.813, R = 0.814, F1: 0.813

With a learning rate of 0.001 and a dropout rate of 0.2, tuned on the validation set.
